core/src/main/java/org/frankframework/http/authentication/AbstractAuthenticator.java

```diff
@@ -0,0 +1,65 @@
+/*
+   Copyright 2024 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package org.frankframework.http.authentication;
+
+import jakarta.annotation.Nullable;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.message.BasicHeader;
+
+import org.frankframework.http.AbstractHttpSession;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.util.List;
+
+import static org.frankframework.util.StreamUtil.DEFAULT_INPUT_STREAM_ENCODING;
+
+public abstract class AbstractAuthenticator implements IOauthAuthenticator {
```

Comment:
Ik zie niet wat deze abstracte class zonder abstract methods toevoegt. Kunnen onderstaande methods misschien als default methods in de interface?

Reply:
Mijn originele plan was om de `SamlAssertionOauth` hiervan te laten inheriten ipv `AbstractOauthAuthenticator`. Dit bleek uiteindelijk een slecht plan, dus vandaar dat deze "nutteloze" class bestaat. Ik zal hem verwijderen!